File: knowledge-hub/knowledge-hub.WebAPI/Database/Book.cs
using System.ComponentModel.DataAnnotations;

namespace knowledge_hub.WebAPI.Database
{
   public class Book
   {
      [Key]
      public int BookId { get; set; }

      [Required]
      public string Name { get; set; }

      [Required]
      public string Author { get; set; }

      [Required]
      public string Description { get; set; }


      public string ImagePath { get; set; }

      [Required]
      public string FilePath { get; set; }

      public double PricePhysical { get; set; }

      public double PriceDigital { get; set; }

      public double Score { get; set; }

      public int LanguageId { get; set; }
      public virtual Language language { get; set; }

      public int CategoryId { get; set; }
      public virtual Category category { get; set; }

   }
}

File: knowledge-hub/knowledge-hub.Mobile/lib/models/book.dart
import 'dart:convert';

import 'comment.dart';

class Book {
  int BookId = 0;
  String Author = "";
  String BookName = "";
  String Language = "";
  String Category = "";
  double PriceDigital = 0;
  double PricePhysical = 0;
  String Description = "";
  double Rating = 5;
  String ImagePath = "";
  String FilePath = "";
  List<Comment> comments = List<Comment>.empty(growable: true);

  Book();

  Map<String, dynamic> toJson() => {
        'bookId': BookId,
        'author': Author,
        'bookName': BookName,
        'language': Language,
        'category': Category,
        'priceDigital': PriceDigital,
        'pricePhysical': PricePhysical,
        'description': Description,
        'score': Rating,
        'imagePath': ImagePath,
        'filePath': FilePath,
        'reviews': comments,
      };
  Book.fromJson(Map<String, dynamic> jsonFile)
      : BookId = jsonFile['bookId'] ?? 0,
        Author = jsonFile['author'] ?? "",
        BookName = jsonFile['name'] ?? "",
        Language = jsonFile['language'] ?? "",
        Category = jsonFile['category'] ?? "",
        PriceDigital = jsonFile['priceDigital'] is int
            ? (jsonFile['priceDigital'] as int).toDouble()
            : jsonFile['priceDigital'],
        PricePhysical = jsonFile['pricePhysical'] is int
            ? (jsonFile['pricePhysical'] as int).toDouble()
            : jsonFile['pricePhysical'],
        Description = jsonFile['description'] ?? "",
        ImagePath = ("http://10.0.2.2:5000/BookImages/") +
            (jsonFile['imagePath'] ?? "default.jpg"),
        FilePath =
            ("http://10.0.2.2:5000/BookFiles/") + (jsonFile['filePath'] ?? ""),
        Rating = jsonFile['score'] is int
            ? (jsonFile['score'] as int).toDouble()
            : jsonFile['score'],
        comments = List<Comment>.empty(growable: true);
}
